Code of Conduct (#586)
drand/drand: CODE_OF_CONDUCT.md

## Consequences

Unacceptable behavior from any community member will not be tolerated.

Anyone asked to stop unacceptable behavior is expected to comply immediately.

If a community member engages in unacceptable behavior, the team may take any action they deem appropriate, up to and including a temporary ban or permanent expulsion from the community without warning (and without refund in the case of a paid event or service).

## Addressing Grievances

If you feel you have been falsely or unfairly accused of violating this Code of Conduct, you should notify the team. We will do our best to ensure that your grievance is handled appropriately.

In general, we will choose the course of action that we judge as being most in the interest of fostering a safe and friendly community.

On IRC, let one of the ops know if you think that someone has transgressed against the Code of Conduct. If you would like to be an op and promise to help maintain and abide by the code, please let us know.

## Contact Info

> Please contact abuse@drand.love if you need to report a problem or address a grievance related to an abuse report.

Reviewer: ```suggestion
Please contact abuse@drand.love if you need to report a problem or address a grievance related to an abuse report. Currently, this address inbox is handled by [Protocol Labs](https://protocol.ai).
```

Reviewer: @willscott good?

Author: i meant more that as in https://github.com/ipfs/community/blob/master/code-of-conduct.md#contact-info it's generally expected that the people who are responsible for handling problems are listed.

# Community Code of Conduct

We believe that our mission is best served in an environment that is friendly, safe, and accepting, and free from intimidation or harassment. Towards this end, certain behaviors and practices will not be tolerated.

## tl;dr

- Be respectful.

> - We're here to help: abuse@ipfs.io

Reviewer: ```suggestion
- We're here to help: abuse@drand.love
```

## Consequences

Unacceptable behavior from any community member will not be tolerated.

Anyone asked to stop unacceptable behavior is expected to comply immediately.

If a community member engages in unacceptable behavior, the team may take any action they deem appropriate, up to and including a temporary ban or permanent expulsion from the community without warning (and without refund in the case of a paid event or service).

## Addressing Grievances

If you feel you have been falsely or unfairly accused of violating this Code of Conduct, you should notify the team. We will do our best to ensure that your grievance is handled appropriately.

In general, we will choose the course of action that we judge as being most in the interest of fostering a safe and friendly community.

On IRC, let one of the ops know if you think that someone has transgressed against the Code of Conduct. If you would like to be an op and promise to help maintain and abide by the code, please let us know.

## Contact Info

> Please contact abuse@ipfs.io if you need to report a problem or address a grievance related to an abuse report.
> The Drand team currently defers to the [IPFS abuse team](https://github.com/ipfs/community/blob/master/code-of-conduct.md#contact-info) for Code of Conduct resolutions. We believe they are best positioned to act as fair mediators with authority over our community for this purpose.

Reviewer: ```suggestion
Please contact abuse@drand.love if you need to report a problem or address a grievance related to an abuse report.
```

# Community Code of Conduct

We believe that our mission is best served in an environment that is friendly, safe, and accepting, and free from intimidation or harassment. Towards this end, certain behaviors and practices will not be tolerated.

## tl;dr

- Be respectful.

> - We're here to help: abuse@ipfs.io

Author: cc @daviddias

> - We're here to help: abuse@ipfs.io

Author: See my comment below. I think it may be better for us to not have our own email address for this

## Consequences

Unacceptable behavior from any community member will not be tolerated.

Anyone asked to stop unacceptable behavior is expected to comply immediately.

If a community member engages in unacceptable behavior, the team may take any action they deem appropriate, up to and including a temporary ban or permanent expulsion from the community without warning (and without refund in the case of a paid event or service).

## Addressing Grievances

If you feel you have been falsely or unfairly accused of violating this Code of Conduct, you should notify the team. We will do our best to ensure that your grievance is handled appropriately.

In general, we will choose the course of action that we judge as being most in the interest of fostering a safe and friendly community.

On IRC, let one of the ops know if you think that someone has transgressed against the Code of Conduct. If you would like to be an op and promise to help maintain and abide by the code, please let us know.

## Contact Info

> The Drand team currently defers to the IPFS abuse team for Code of Conduct resolutions. Please contact abuse@ipfs.io if you need to report a problem or address a grievance related to an abuse report.

Reviewer 2: Just have an abuse@drand? I feel we should try to prioritise getting help in this document not who is providing it. Personally I’d remove references to IPFS here.

Author: IPFS is large enough to have trained an abuse team to handle these issues, and can act as a neutral 3rd party arbiter. we could do much worse than deferring to that abuse team if us an them are willing to have that happen.

Author: when there are code of conduct issues, it's important that the person sending the report knows who it's going to, and has transparency in that. Ideally we list the people here, but if we defer to IPFS i think it's okay to include a link to the IPFS code of conduct so that there's one authoritative source of that membership rather than this one potentially getting stale. (re your pointer to credit below)

Reviewer 2: > IPFS is large enough to have trained an abuse team to handle these issues, and can act as a neutral 3rd party arbiter. we could do much worse than deferring to that abuse team if us an them are willing to have that happen.

Um, okay. Just to be clear I was not suggesting that it’s a bad idea to defer this to the IPFS abuse team or that they are in any way a bad choice for it.

My thoughts were just to ensure the contact info is front and centre, instead of an advertisement for IPFS, which is how I felt it could come across.

In hindsight I think you’re right and if we’re going to defer to IPFS for abuse issues then for transparency we actually _shouldn’t_ have our own email address.

In which case we can easily resolve this by switching the sentence around so that the contact details come first, followed by clarification of who is receiving it and why.

We will keep all matters confidential and they will only be shared within the team, with the exception of legal counsel when deemed necessary, or unless given explicit permission by the reporter.

You are also encouraged to contact us if you are curious about something that might be "on the line" between appropriate and inappropriate content. We are happy to provide guidance to help you be a successful part of our community.

## Changes

This is a living document and may be updated from time to time. Please refer to the git history for this document to view the changes.

> ## Credit and License

Reviewer 2: This feels like a better place to say who will be dealing with the help requests (IPFS abuse).
